### ISSUE TITLE
Suggestion for the future: RetroAchievements support

### ISSUE THREAD

**Issue by nighto** [hikari-no-yume/touchHLE#55]
I'd like to give a suggestion for the future. There is a very nice project called [RetroAchievements](https://retroachievements.org/), it's a community that adds achievements to emulated games - similar to what you'd have on PlayStation 3+ or Xbox. Basically create achievements to Super Monkey Ball when, say, you finish the tutorial, finish a certain number of levels, finish the levels without dying, finish the game etc etc.

Of course there are a loooot of things to work on the near future but I personally would love to see it on the radar. =)

RetroAchievements Developers documentation: https://docs.retroachievements.org/Developer-docs/

**Comment by hikari-no-yume** [comment]
Hmm, adding achievements for games that don't have built-in support for them seems like a complicated enhancement. Maybe we can eventually have a plugin system of some kind so that people can create achievements systems if they want to.

On the other hand, eventually I am going to have to implement the Game Center API, which was Apple's official way to do achievements in iOS apps. I don't know how that works, but I can imagine a service like RetroAchievements being useful there.

**Comment by nighto** [comment]
Yes, that's exactly what RA is all about. Basically reading memory addresses and doing some sort of reverse engineering to figure out where the program stores the data etc.

A link if you are curious how it is done: https://docs.retroachievements.org/Getting-Started-as-an-Achievement-Developer/

But yeah, interfacing Game Center API and forwarding standard achievements to Retro Achievements would be also awesome :)
